src/AudioPlayer.cpp

```diff
@@ -1067,21 +1060,17 @@ size_t AudioPlayer_NvsRfidWriteWrapper(const char *_rfidCardId, const char *_tra
 
 // Adds webstream to playlist; same like SdCard_ReturnPlaylist() but always only one entry
 char **AudioPlayer_ReturnPlaylistFromWebstream(const char *_webUrl) {
-	char *webUrl = x_strdup(_webUrl);
-	static char **url;
+	static char number[] = "1";
+	static char *url[2] = {number, NULL};
 
-	if (url != NULL) {
-		--url;
-		freeMultiCharArray(url, strtoul(*url, NULL, 10));
+	if(url[1] != NULL) {
+		free(url[1]);
 	}
 
-	url = (char **)x_malloc(sizeof(char *) * 2);
-
-	url[0] = x_strdup("1"); // Number of files is always 1 in url-mode
-	url[1] = x_strdup(webUrl);
+	number[0] = '1';
+	url[1] = x_strdup(_webUrl);
 
-	free(webUrl);
-	return ++url;
+	return &(url[1]);
 }
```

Comment:
This function comes from hell ;)
What is it supposed to do? Your code looks better than the original version, but it still causes a headache when I try to read it. I'm pretty sure this can be simplified, but I need to understand what the code is supposed to do first. A comment in the code would also help a lot.

Reply:
Yes I know, the function looks like a crutch ("Ne Krücke") :( 

The problem stems from the usage of the char** array (the first element is used to hold the number of entries (in ASCII text). This layout is currently used to create the playlist from different sources and transfer it to the AudiPlayer Task. For the webstream (which always has only a single entry) this creates this memory layout:

```
                             strdup
              malloc           │
                │              ▼
 char**         ▼           ┌───────┐
┌─────┐     ┌────────┐   ┌─►│ "1\0" │
│ url ├────►│ url[0] ├───┘  └───────┘
└─────┘     ├────────┤
            │ url[1] ├───┐
            └────────┘   │  ┌──────────────┐
                         └─►│     <url>    │
                            └──────────────┘
                                    ▲
                                    │
                                 strdup
```



What I tried was to put the pointer array and the "1" into the global memory instead of the heap. This was the best I could come up short of rewriting playlist management thing (I wanted to keep this PR relatively small).

I was playing around with the idea of changing the return value of the functions to a struct with a char** and a counter (or an std::vector of unique_ptr and be done with it :) ). This would need a deeper rework of multiple functions, at least:
- AudioPlayer_ReturnPlaylistFromWebstream
- AudioPlayer_TrackQueueDispatcher
- SdCard_ReturnPlaylist